src/main/java/org/folio/search/utils/ShelvingOrderCalculationHelper.java

```diff
@@ -0,0 +1,19 @@
+package org.folio.search.utils;
+
+import lombok.NonNull;
+import lombok.experimental.UtilityClass;
+import org.folio.search.domain.dto.ShelvingOrderAlgorithmType;
+import org.marc4j.callnum.DeweyCallNumber;
+import org.marc4j.callnum.LCCallNumber;
+
+@UtilityClass
+public class ShelvingOrderCalculationHelper {
+
+  public static String calculate(@NonNull String input, @NonNull ShelvingOrderAlgorithmType algorithmType) {
+    return switch (algorithmType) {
+      case LC -> new LCCallNumber(input).getShelfKey().trim();
+      case DEWEY -> new DeweyCallNumber(input).getShelfKey().trim();
+      case DEFAULT -> CallNumberUtils.normalizeEffectiveShelvingOrder(input);
```

Comment:
I don't think this is acceptable to use as internally it has "isSupportedCharacter" which will replace characters not supporter by call number numeric algorithm